src/output/workspace-impl.cpp

```diff
@@ -230,13 +230,23 @@ struct workspace_set_t::impl
             auto wm  = view->get_geometry();
             float px = 1. * wm.x / old_w;
             float py = 1. * wm.y / old_h;
-            float pw = 1. * wm.width / old_w;
-            float ph = 1. * wm.height / old_h;
 
-            view->set_geometry({
-                    int(px * new_geometry.width), int(py * new_geometry.height),
-                    int(pw * new_geometry.width), int(ph * new_geometry.height)
-                });
+            auto edges = view->toplevel()->current().tiled_edges;
+            if (view->get_output() && view->toplevel()->current().fullscreen)
+            {
+                wf::get_core().default_wm->fullscreen_request(view, view->get_output(), false);
```

Comment:
This feels a bit hacky, I would directly adjust the view geometry for fullscreen views.

Reply:
You mean to say, resize the view directly?
```
view->set_geometry(...)
```

Reply:
Yes